elasticsearch/src/main/java/io/zipkin/dependencies/spark/elasticsearch/ElasticsearchDependenciesJob.java

```diff
@@ -0,0 +1,156 @@
+        "spark.ui.enabled", "false",
+        // avoids strange class not found bug on Logger.setLevel
+        "spark.akka.logLifecycleEvents", "true",
+        // don't die if there are no spans
+        "es.index.read.missing.as.empty", "true",
+        "es.nodes", getEnv("ES_HOSTS", "127.0.0.1") // ES_HOSTS is the env variable in zipkin
+    );
+
+    // local[*] master lets us run & test the job locally without setting a Spark cluster
+    String sparkMaster = getEnv("SPARK_MASTER", "local[*]");
+
+    // By default the job only works on traces whose first timestamp is today
+    long day = midnightUTC(System.currentTimeMillis());
+
+    /** The index prefix to use when generating daily index names. Defaults to "zipkin" */
+    public Builder index(String index) {
+      this.index = checkNotNull(index, "index");
+      return this;
+    }
+
+    /** Day (in epoch milliseconds) to process dependencies for. Defaults to today. */
+    public Builder day(long day) {
+      this.day = midnightUTC(day);
+      return this;
+    }
+
+    public ElasticsearchDependenciesJob build() {
+      return new ElasticsearchDependenciesJob(this);
+    }
+
+    Builder() {
+    }
+  }
+
+  final String index;
+  final long day;
+  final String dateStamp;
+  final SparkConf conf;
+
+  static final JsonAdapter<TraceId> traceIdReader =
+      new Moshi.Builder().build().adapter(TraceId.class);
+
+  static final class TraceId {
+    String traceId;
+  }
+
+  ElasticsearchDependenciesJob(Builder builder) {
+    this.index = builder.index;
+    this.day = builder.day;
+    this.dateStamp = new SimpleDateFormat("yyyy-MM-dd").format(new Date(builder.day));
+    this.conf = new SparkConf(true)
+        .setMaster(builder.sparkMaster)
+        .setAppName(getClass().getName());
+    for (Map.Entry<String, String> entry : builder.sparkProperties.entrySet()) {
+      conf.set(entry.getKey(), entry.getValue());
+    }
+  }
+
+  public void run() {
+    String bucket = index + "-" + dateStamp;
+
+    System.out.println("Processing spans from " + bucket + "/span");
+
+    JavaSparkContext sc = new JavaSparkContext(conf);
+    JavaRDD<Map<String, Object>> links = JavaEsSpark.esJsonRDD(sc, bucket + "/span")
+        .groupBy(pair -> traceIdReader.fromJson(pair._2).traceId)
+        .flatMap(pair -> toLinks(pair._2))
+        .mapToPair(link -> Tuple2.apply(Tuple2.apply(link.parent, link.child), link))
+        .reduceByKey((l, r) -> DependencyLink.create(l.parent, l.child, l.callCount + r.callCount))
+        .values()
+        .map(l -> ImmutableMap.<String, Object>of(
+            "parent", l.parent,
+            "child", l.child,
+            "id", l.parent + "|" + l.child,
+            "callCount", l.callCount
+        ));
+
+    System.out.println("Saving dependency links to " + bucket + "/dependencylink");
+    JavaEsSpark.saveToEs(links, bucket + "/dependencylink",
+        ImmutableMap.of("es.mapping.id", "id")); // unique id, which allows overwriting the link
```

Comment:
would be nice if es.mapping.id took an expression. That way, we'd not have to store "id" just to map it here..